go-controller/hack/build-go.sh

```diff
@@ -29,7 +29,8 @@ build_binaries() {
                 -X ${OVN_KUBE_GO_PACKAGE}/pkg/config.Commit=${GIT_COMMIT} \
                 -X ${OVN_KUBE_GO_PACKAGE}/pkg/config.Branch=${GIT_BRANCH} \
                 -X ${OVN_KUBE_GO_PACKAGE}/pkg/config.BuildUser=${BUILD_USER} \
-                -X ${OVN_KUBE_GO_PACKAGE}/pkg/config.BuildDate=${BUILD_DATE}" \
+                -X ${OVN_KUBE_GO_PACKAGE}/pkg/config.BuildDate=${BUILD_DATE} \
+                -X k8s.io/client-go/pkg/version.gitVersion=v0.20.0" \
```

Comment:
does it uses the absolute or relative path?
some google searches show people using the vendored path

```
-X 'k8s.io/kubernetes/vendor/k8s.io/client-go/pkg/version.gitVersion=v1.20.0-beta.0.1281+87984d84d18c78-dirty'
```

Reply:
It uses relative path. I tried absolute path and it was not working. It really depends on how folks are building the binary.

If they are using go1.13 and below and are building against mod cache, then the gitVersion will not be set.

If they are using go1.14 and above and are building, by default, against vendor, then the absolute path will work.

With relative path, it doesn't matter whether they are building against a mod cache or vendor, it will always work.

Reply:
heh, go modules fun :smile:

Reply:
do we want to hardcode the version to current kubernetes vendored library?
or use the ovn git commit?

Reply:
@aojea we can do both. Let me know if you want to do both? For now k8s version seems like a good start.

Reply:
just asking, don´t have a strong opinion, I´m afraid we forget to bump the version

Reply:
you find the issue so you should have a better opinion, what will provide better feedback in the logs?

> user-agent:ovnkube/v0.20.0 (linux/amd64) kubernetes/$Format

the kubernetes version used by the agent or the ovn one?

Reply:
I would prefer we dont hardcode this:
k8s.io/client-go/pkg/version.gitVersion=v0.20.0"

We are going to forget to bump it when we bump other stuff. Maybe you can grep it out of the go mod or something.